File: app/src/main/java/com/wilinz/yuetingmusic/util/MimeTypes.java
package com.wilinz.yuetingmusic.util;


import android.webkit.MimeTypeMap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.apache.commons.io.FilenameUtils;

/**
 * Created by Stardust on 2018/2/12.
 */

public class MimeTypes {

    @Nullable
    public static String fromFile(String path) {
        String ext = FilenameUtils.getExtension(path);
        return android.text.TextUtils.isEmpty(ext) ? "*/*" : MimeTypeMap.getSingleton().getMimeTypeFromExtension(ext);
    }

    @NonNull
    public static String fromFileOr(String path, String defaultType) {
        String mimeType = fromFile(path);
        return mimeType == null ? defaultType : mimeType;
    }
}

File: app/src/main/java/com/wilinz/yuetingmusic/util/MimeTypes.kt
package com.wilinz.yuetingmusic.util

import android.text.TextUtils
import android.webkit.MimeTypeMap
import org.apache.commons.io.FilenameUtils

/**
 * Created by Stardust on 2018/2/12.
 */
object MimeTypes {
    fun fromFile(path: String?): String? {
        val ext = FilenameUtils.getExtension(path)
        return if (TextUtils.isEmpty(ext)) "*/*" else MimeTypeMap.getSingleton()
            .getMimeTypeFromExtension(ext)
    }

    fun fromFileOr(path: String?, defaultType: String): String {
        val mimeType = fromFile(path)
        return mimeType ?: defaultType
    }
}